apps/web/src/components/tx-flow/flows/CreateNestedSafe/ReviewNestedSafe.tsx

```diff
@@ -0,0 +1,150 @@
+import type { SetupNestedSafeForm } from '@/components/tx-flow/flows/CreateNestedSafe/SetupNestedSafe'
+
+export function ReviewNestedSafe({ params }: { params: SetupNestedSafeForm }): ReactElement {
+  const dispatch = useAppDispatch()
+  const { safeAddress, safe, safeLoaded } = useSafeInfo()
+  const chain = useCurrentChain()
+  const { setSafeTx, setSafeTxError } = useContext(SafeTxContext)
+  const { balances } = useBalances()
+  const provider = useWeb3ReadOnly()
+  const { data: nestedSafes } = useGetOwnedSafesQuery(
+    safeLoaded ? { chainId: safe.chainId, ownerAddress: safeAddress } : skipToken,
+  )
+  const version = getLatestSafeVersion(chain)
+
+  const safeAccountConfig = useMemo(() => {
+    if (!chain || !nestedSafes) {
+      return
+    }
+
+    const undeployedSafe = createNewUndeployedSafeWithoutSalt(
+      version,
+      {
+        owners: [safeAddress],
+        threshold: 1,
+      },
+      chain,
+    )
+    const saltNonce = Date.now().toString()
+
+    return {
+      ...undeployedSafe,
+      saltNonce,
+    }
+  }, [chain, safeAddress, nestedSafes, version])
+
+  const [predictedSafeAddress] = useAsync(async () => {
+    if (provider && safeAccountConfig) {
+      return predictAddressBasedOnReplayData(safeAccountConfig, provider)
+    }
+  }, [provider, safeAccountConfig])
+
+  useEffect(() => {
+    if (!chain || !safeAccountConfig || !predictedSafeAddress) {
+      return
+    }
+
+    const deploymentTx = {
+      to: safeAccountConfig.factoryAddress,
+      data: encodeSafeCreationTx(safeAccountConfig, chain),
+      value: '0',
+    }
+
+    const fundingTxs: Array<MetaTransactionData> = []
+
+    for (const asset of params.assets) {
+      const token = balances.items.find((item) => {
+        return item.tokenInfo.address === asset[SetupNestedSafeFormAssetFields.tokenAddress]
+      })
+
+      if (token) {
+        fundingTxs.push(
+          createTokenTransferParams(
+            predictedSafeAddress,
+            asset[SetupNestedSafeFormAssetFields.amount],
+            token.tokenInfo.decimals,
+            token.tokenInfo.address,
+          ),
+        )
+      }
+    }
+
+    const createSafeTx = async (): Promise<SafeTransaction> => {
+      const isMultiSend = fundingTxs.length > 0
+      return isMultiSend ? createMultiSendCallOnlyTx([deploymentTx, ...fundingTxs]) : createTx(deploymentTx)
+    }
+
+    createSafeTx().then(setSafeTx).catch(setSafeTxError)
+  }, [chain, params.assets, safeAccountConfig, predictedSafeAddress, balances.items, setSafeTx, setSafeTxError])
+
+  const onSubmit = () => {
+    if (!predictedSafeAddress) {
+      return
+    }
+    dispatch(
+      upsertAddressBookEntries({
+        chainIds: [safe.chainId],
+        address: predictedSafeAddress,
+        name: params.name,
+      }),
+    )
+  }
+
+  return (
+    <SignOrExecuteForm onSubmit={onSubmit}>
+      {predictedSafeAddress && (
+        <Grid
```

Comment:
Grid is deprecated, it's better to use Grid2 for new code.

Reply:
Changed in 4b3cd6a762afbbce0d3bd543605654e8f8578b3f.